src/Kernel/Object.class.st

```diff
@@ -140,6 +140,17 @@ Object class >> newFrom: aSimilarObject [
 		ifFalse: [self basicNew]) copySameFrom: aSimilarObject
 ]
 
+{ #category : #'instance creation' }
+Object class >> newFromArray: anArray [
```

Comment:
Do we want this on Object?
I would not.

Reply:
As the comment of the method says, this is only for backward compatibility.

Some existing clients can (directly or not) rely on the fact that `{1. 2. 3} as: SomeClass` calls back `SomeClass>>newFrom:`. With the change proposed in the PR the likely DNU selector `newFromArray:` will be called instead. So to have a semantically equivalent change, I just globally forward `newFromArray:` to `newFrom:` unless a better `newFromArray:` is redefined.

Side note: I'm not sure about the *backward compatibility policy* of Pharo (is there an implicit or an explicit one?) I can adapt the PR to what you prefer: I can drop `Object class>>newFromArray:` (and zut for guys that are on the way). I can also harden the current proposal for compatibility: move specific behavior Collection>>newFromArray into selected subclasses (so weird subclasses of Collection that redefine newFrom: or withAll: will not break). I can even add type guards to that weird subclasses of OrderedCollection or Array do not feel the change (fragile base class bla bla bla).

Other note: the rationale and laws of Object-level methods like `Object>>as:` or `Object class>>newFrom:` are not completely evident for me. While this is not the point of the PR to discuss these, feel free to say if I broke some beahviors or rules.